File: codes/3-front-end/day-2/payroll-app/src/models/developer.ts
import { Employee } from "./employee";

export class Developer extends Employee {
    constructor(id: number, name: string, basic: number, da: number, hra: number, public incentivePayment: number) {
        super(id, name, basic, da, hra)
    }
    calculateSalary() {
        super.calculateSalary()
        this.totalSalary += this.incentivePayment
    }
}

File: codes/2-java/casestudy/emsapp/src/oracle/java/emsapp/entities/Developer.java
package oracle.java.emsapp.entities;

public class Developer extends Employee {

	private double incentivePayment;

	public Developer() {

	}

	public Developer(String name, int id, double basicPayment, double daPayment, double hraPayment,
			double incentivePayment) {
		super(name, id, basicPayment, daPayment, hraPayment);
		this.incentivePayment = incentivePayment;
	}

	public double getIncentivePayment() {
		return incentivePayment;
	}

	public void setIncentivePayment(double incentivePayment) {
		this.incentivePayment = incentivePayment;
	}

	@Override
	public void calculateSalary() {
		super.calculateSalary();
		this.setTotalPayment(this.getTotalPayment() + this.incentivePayment);
	}
}
